### DISCUSSION TITLE
Setup question for Nextcloud tasks via DAVx5

### DISCUSSION THREAD

**Discussion by foss-** [TechbeeAT/jtxBoard#617] [Q&A]
Failing to setup jtx board for Nextcloud tasks sync via DAVx5. The setup in DAVx5 4.3.3-ose seems fine, since I sync tasks to Tasks 13.2.4, interact / update tasks and that info syncs forth and back.

Re-read https://jtx.techbee.at/sync-with-davx5. Checked App Permissions which look fine and are all enabled. `jtx Board permissions` is enabled as is `Tasks permissions`.

When I do a sync, I see no tasks at all. When adding a new task in jtx Board that is created in `Local (Local)` by default. I can see some Synology CalDAV Tasks lists that exist in DAVx5 and select them, but the Nextcloud entry in DAVx5 does not show in jtx Board when I look at `Settings > Collections`. If I tap the top right `… > Add remote collection (DAVx5), that does open DAVx5. But I am unsure what to do. https://jtx.techbee.at/sync-with-davx5 says "If your CalDAV-server supports VJOURNAL, make sure that you have either an existing collection with VJOURNAL support – or create a new collection for an account with VJOURNAL support enabled." But I am unclear what is missing or which steps to take. https://www.davx5.com/tested-with/nextcloud isn't helpful either.

There are various issues about Nextcloud supporting VJOURNAL, which it seems to not do by default. Then https://github.com/nextcloud/calendar/issues/1514#issuecomment-1079996768 mentions jtx Board and DAVx5 which allows to create a collection to enable VJOURNAL for a nextcloud calendar. But I can't seem to find documentation covering this in jtx Board, Nextcloud or DAVx5.

Out of ideas what to try or what I am doing wrong. This is DAVx5:
<img src="https://github.com/TechbeeAT/jtxBoard/assets/5708172/6f1d5e3f-2663-4edb-a9f1-e50462d3790a" width="300">

**Comment by patrickunterwegs** [comment]
Hi @foss- , please also have a look at this article: 
https://jtx.techbee.at/troubles-with-the-synchronization-see-what-could-go-wrong

Basically tasks should synchronise, for journals you would need to make a new collection/calendar through the top menu and make sure that the journal checkbox is set. Also you need to make sure that jtx Board is set as your preferred tasks app in the DAVx5 settings. Did this solve your problem?

**Comment by foss-** [comment]
@patrickunterwegs thanks so much for your reply. I had found that article but was unable to get to a state in which in DAVx5 settings, one entry shows all three icons as per the documentation screenshot. Which option do I have to change for that?

In DAVx5 settings I changed `Settings > Integration > Tasks support` from Tasks.org to jtx Board. After that, tasks indeed show up in jtx Board. So my misunderstanding was, that I would be able to use both task apps simultaneously, but apparently that is not the case. Does this indeed mean, tasks.org will now no longer sync when no longer set as default task app? Maybe that info could be added to https://jtx.techbee.at/sync-with-davx5? It's funny, that it seems I am the first user to run into this. The fact that DAVx5 does have an ON | OFF toggle as well as a selection at the front is confusing. If I only can sync with one app (the one selected in front), what does the toggle then do?

Re: enabling VJOURNAL for my Nextcloud Calendar, I now have a pending request about that and will report back once that is resolved.

**Reply by patrickunterwegs** [reply]
@foss- regrettably you can't change a collection to activate journals support for a remote collection through DAVx5. But you can create a new collection (calendar), see the section "Make sure you have at least one collection that supports tasks or journals" in https://jtx.techbee.at/troubles-with-the-synchronization-see-what-could-go-wrong

Yes, currently you can choose only one app to synchronize your tasks and journals with. For your understanding: Andorid has a local calendar storage and DAVx5 synchronises calendar entries to this storage. Like that you can access your calendar entries with multiple apps, as all of them access the local calendar storage. But there is no central tasks or journals storage in Android, so apps that use this data structure must implement their own database/storage. So journals and tasks are synchronised directly to other apps and currently only one app for tasks and journals is supported by DAVx5.

With the toggle you can just remove collections from the sync (while keeping it in the account).

But basically journals support should already be given in Nextcloud, just make sure you create a new collection!

**Reply by foss-** [reply]
After the hoster of my nextcloud edited `oc_calendars` I now see the journal symbol. And finally after creating another new collection all three icons show. 🙌

Are you sure about only being able to sync one app? I switched to jtx as main app but tasks app is still receiving updates. I was not able to use the toggle in DAVx5 to disable tasks app and since notifications still showed for tasks app I removed the app to avoid confusion.

So for my understanding, I added a journal test entry. Should that journal entry show in the Calendar of that collection?

And for notes, is there some Nextcloud UI for that? It seems the notes in Nextcloud are file based and an entirely different thing than the notes section in jtx synching via CalDAV. Do you happen to know if there is a way to access journals and notes from macOS?

**Reply by patrickunterwegs** [reply]
Hey @foss- , 
the current architecture of DAVx5 should not allow multiple tasks apps. But there have been some recent changes there as they switched to the Work Manager, I'll double check if I can see the same behaviour.

A collection/calendar can contain journals/notes, tasks and events. Only events would be visible in other calendars on your device.

Regrettably there is no UI yet for notes and journals in other apps or web applications, at least not that I know of. The idea behind jtx Board was also to provide an app that supports those components so that other developers might do the same and create more applications that support the standard. Regrettably that hasn't happened so far (at least not that I know of).

**Reply by rogercreagh** [reply]
I think you can set up tasks.org app to sync using CalDAV directly OR Davx5 - so you can get round the Davx5 single task app limitation by using the native tasks.org CalDAV client?

**Reply by rogercreagh** [reply]
> Regrettably that hasn't happened so far (at least not that I know of).

Yet - give it time....

**Comment by rogercreagh** [comment]
Just to add re Vjournal in NextCloud, it is possible to add Vjournal support to an existing calendar, but not from within jtxB (or any other app). It involves having to change a value directly in the Nextcloud database using phpMyAdmin or similar tool - details see below. This obviously requires that you have access to the database on the server - which is ok if you are running your own Nextcloud server but probably not so easy if you are using third party (or Nextcloud's own paid-for) hosting.

Unfortunately the CalDAV spec [rfc4791](https://www.rfc-editor.org/rfc/rfc4791#page-14) specifies that the supported components property (VEVENT, VTODO,VJOURNAL, etc) for a calendar can **not** be changed by a client, but only on the server. Clients can only create a new calendar with whatever combination of components they require. So there is no possibility of an app being able to remotely change the status of an existing calendar.  

Equally unfortunately Nextcloud does not currently provide any way to either include VJOURNAL as a component when adding a calendar, or to add VJOURNAL component support to an existing calendar. See [this issue](https://github.com/nextcloud/calendar/issues/4436) on Nextcloud GitHub - and add your support if you've got a github account.

Without database access on the server all you can do is create a new calendar with VJOURNAL support. Of course this is a pain if you are wanting to integrate journals into an existing user's calendar.

**IF** you have phpMyAdmin (or similar MySQL database fiddling tool) available on your calendar server all you have to do is open the Nextcloud database and find the table `oc_calendars`. This lists all of the calendars available on the server. Find the row for the calendar you wish to add VJOURNAL support to and edit the field `components` to add `,VJOURNAL` to whatever is there already.  NB uppercase, with a comma between each component type and no spaces.

**Reply by foss-** [reply]
Also this open feature request https://github.com/nextcloud/tasks/issues/1955
NextCloud has a fascinating argument to not enable VJOURNAL by default:
> I don't think this makes sense as long as nothing in the Nextcloud ecosystem makes use of VJOURNAL and that it's not widely used anywhere at the moment. https://github.com/nextcloud/calendar/issues/4436#issuecomment-1226810894

Don't think that is a convincing argument. Oh well...
Have a pending support request with my hoster to edit `oc_calendars`.
